tests/components/ruckus_unleashed/test_device_tracker.py

```diff
@@ -101,3 +112,24 @@ async def test_restoring_clients(hass):
     device = hass.states.get(TEST_CLIENT_ENTITY_ID)
     assert device is not None
     assert device.state == STATE_NOT_HOME
+
+
+async def test_client_device_setup(hass):
+    """Test a client device is created."""
+    await init_integration(hass)
+
+    router_info = DEFAULT_AP_INFO[API_AP][API_ID]["1"]
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    client_device = device_registry.async_get_device(
+        identifiers={},
+        connections={(CONNECTION_NETWORK_MAC, TEST_CLIENT[API_MAC])},
+    )
+    router_device = device_registry.async_get_device(
+        identifiers={(CONNECTION_NETWORK_MAC, router_info[API_MAC])},
+        connections={(CONNECTION_NETWORK_MAC, router_info[API_MAC])},
+    )
+
+    assert client_device
+    assert client_device.name == TEST_CLIENT[API_NAME]
+    assert client_device.via_device_id == router_device.id
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)